clang/include/clang/Basic/Attr.td

```diff
@@ -4723,20 +4723,25 @@ def HLSLResourceBinding: InheritableAttr {
 
   private:
       RegisterType RegType;
-      unsigned SlotNumber;
+      int SlotNumber; // -1 if the register slot was not specified
```

Comment:
I don't think we can do this - what about `RWBuffer<float4> B : register(2147483648)` or `RWBuffer<float4> C : register(42949672925)`? `dxc` accepts any register that fits in unsigned as far as I am aware.

Better to use `std::optional<int>` here.